### ISSUE TITLE
Integration with Flask-Twisted Library

### ISSUE THREAD

**Issue by nickh4nson** [miguelgrinberg/Flask-SocketIO#135]
I am currently building an application using the [Twisted](https://github.com/twisted/twisted) library and  employing the [Flask-Twisted](https://github.com/cravler/flask-twisted) library to tie into my framework.

Flask-Twisted is setup very similarly to SocketIO
```python 
twisted = Twisted(app)
socketio = SocketIO(app) 
```

but will only work when I use
```python 
app.run(port=5000)
```

and does not work with
```python 
socketio.run(app, port=5000)
```

Is there some configuration or small change I can make to address this issue?

**Comment by miguelgrinberg** [comment]
It looks like Flask-Twisted monkey patches `app.run`, that explains why that's the only way you can make it work.

I'm not sure you'll be able to make the two work together. The starting of the server is easy to address, but Flask-SocketIO uses gevent (or alternatively eventlet if you are using the 1.0a1 development version). I'm not super familiar with Twisted, but it seems to me the extension will not be able to use gevent/eventlet when running with Twisted.

With the upcoming 1.0 release it might be possible to add support for Twisted, so in the future this may be a viable combination.

**Comment by nickh4nson** [comment]
Thank you for the quick reply.

This sounds like a more fundamental problem of running a sockets server and TCP client in one instance of Flask. I still need the web sockets functionality for my application. If you don't mind me asking, should I look at setting up a separate server specifically for sending socket messages to the client or should I try out other libraries that doesn't use gevent?

**Comment by miguelgrinberg** [comment]
@nickh4nson Do you need to use Twisted? If you can use gevent or eventlet for your WSGI server, then you can have both the HTTP and socket parts of your application together.

**Comment by nickh4nson** [comment]
The application will be updating configuration based on messages sent over a TCP connection from another server, and to listen to the connection I have found Twisted to be the best supported library.

On other projects I have used your library to successfully run the HTTP and sockets parts together. I ask about breaking out sockets portion for the current application because I can see not directly updating configuration through the sockets connection and just using it to bring the client up to date.
